server/main.py

```diff
@@ -162,9 +162,10 @@ def api_get_applicant_list():
     return jsonify(get_applicant_list())
 
 
-@app.route('/api/user/<int:user_id>/characters')
+@app.route('/api/user/characters')
+@app.route('/api/user/characters/<int:character_id>')
```

Comment:
Change character_id to user_id and target_char_id=None to user_id=None and this is good.